src/app/pages/individual-condo/individual-condo.component.ts

```diff
@@ -1,97 +1,181 @@
-    NumberOfBedrooms: 2,
-    NumberOfBathrooms: 1,
-    SquareFootage: 1000,
-    ID: '',
-    Type: CondoType.Sale,
-    OccupantID: '',
-    Status: CondoStatus.Vacant,
-  };
-
-  userInfo: UserDTO = {
-    ProfilePicture:
-      'https://slidesbase.com/wp-content/uploads/2023/07/03-happy-young-latin-real-estate-agent-woman-stock-photo_slidesbase-1.jpg',
-    FirstName: 'Maria',
-    LastName: 'Lapa',
-    Email: 'marialapa@gmail.com',
-    PhoneNumber: '+1 (514)-567-8901',
-    Authority: Authority.Public,
-    ID: '1234',
-    UserName: 'marialapa',
-    Notifications: [
-      {
-        Message: 'Your request has been approved.',
-        Date: new Date().getTime(),
-        New: true,
-        SenderId: '1234',
-      },
-      {
-        Message: 'Your request has been approved.',
-        Date: new Date().getTime(),
-        New: true,
-        SenderId: '1234',
-      },
-      {
-        Message: 'Your request has been approved.',
-        Date: new Date().getTime(),
-        New: true,
-        SenderId: '1234',
-      },
-    ],
-  };
+  async ngOnInit(): Promise<void> {
+    this.route.params.subscribe(async (params) => {
+      const buildingId = params['buildingId'];
+      const condoId = params['condoId'];
+      await this.fetchBuildingAndCondoDetails(buildingId, condoId);
+      await this.fetchUserInfo();
+      this.loggedInUser = await this.fetchLoggedInUser();
+      // this.myUser = await this.authService.getUser();
+    });
+  }
 
-  loading = false;
+  async fetchBuildingAndCondoDetails(
+    buildingId: string,
+    condoId: string
+  ): Promise<void> {
+    try {
+      const buildingDetails = await this.buildingService.getBuilding(
+        buildingId
+      );
+      if (buildingDetails) {
+        this.building = buildingDetails;
+        // Now find the specific condo in the building's condos array
+        this.condo =
+          this.building.Condos?.find((condo) => condo.ID === condoId) || null;
+        if (!this.condo) {
+          console.error('Condo not found in the building');
+        }
+      } else {
+        console.error('Building details not found');
+      }
+    } catch (error: any) {
+      console.error(
+        'Error fetching building and condo details:',
+        error.message
+      );
+    }
+  }
 
-  goBack() {
-    console.log('Functionality not implemented yet.');
+  async fetchUserInfo(): Promise<void> {
+    if (this.condo?.Status === 'Vacant') {
+      this.userInfo = await this.userService.getCompanyUser(
+        this.building?.CompanyID || ''
+      );
+    } else {
+      this.userInfo = await this.userService.getPublicUser(
+        this.condo?.OccupantID || ''
+      );
+      if (!this.userInfo) {
+        this.userInfo = await this.userService.getEmployeeUser(
+          this.condo?.OccupantID || ''
+        );
+      }
+    }
   }
 
+  goBack(): void {
+    // Navigate to previous route
+    this.router.navigateByUrl('./condos');
```

Comment:
Maybe use Angular's Location service to go back to the previous page. Here's how to use it:
`import { Location } from '@angular/common';`
`constructor(private location: Location) { }`

`goBack(): void {
  this.location.back();
}`